tutormfe/plugin.py

```diff
@@ -90,12 +127,21 @@ def get_mfes() -> dict[str, MFE_ATTRS_TYPE]:
     return MFE_APPS.apply({})
 
 
+@functools.lru_cache(maxsize=None)
+def get_slots() -> list[str]:
+    """
+    This function is cached for performance.
+    """
+    return MFE_SLOTS.apply([])
+
+
 @tutor_hooks.Actions.PLUGIN_LOADED.add()
 def _clear_get_mfes_cache(_name: str) -> None:
     """
     Don't forget to clear cache, or we'll have some strange surprises...
     """
     get_mfes.cache_clear()
+    get_slots.cache_clear()
```

Comment:
Let's not do that. This will break in some edge cases. Instead, both `get_slots` and `get_mfes` should use [tutor.hooks.lru_cache](https://docs.tutor.edly.io/reference/api/hooks/index.html#tutor.hooks.lru_cache).